### ISSUE TITLE
Bug: sam deploy - AttributeError

### ISSUE THREAD

**Issue by mikegrima** [aws/aws-sam-cli#4937]
### Description:
Just upgraded to the SAM CLI version 1.78.0, and am now seeing this error on deployment:

```
% sam deploy --config-env TEST      

Error: 'str' object has no attribute 'value'
Traceback:
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/samcli/lib/cli_validation/image_repository_validation.py", line 92, in wrapped
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/samcli/lib/telemetry/metric.py", line 183, in wrapped
    raise exception  # pylint: disable=raising-bad-type
    ^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/samcli/lib/telemetry/metric.py", line 148, in wrapped
    return_value = func(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/samcli/lib/utils/version_checker.py", line 42, in wrapped
    actual_result = func(*args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/samcli/cli/main.py", line 90, in wrapper
    cmdline_args_log += f"--{key}={str(value)} "
                                   ^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/tomlkit/items.py", line 1415, in __str__
    return str([v.value.value for v in self._iter_items() if v.value is not None])
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.78.0/libexec/lib/python3.11/site-packages/tomlkit/items.py", line 1415, in <listcomp>
    return str([v.value.value for v in self._iter_items() if v.value is not None])
                ^^^^^^^^^^^^^
```

Didn't see this before on the older versions.  Below is a sample of my `samconfig.toml`:

```toml
version = 0.1
[TEST]
[TEST.deploy]
[TEST.deploy.parameters]
stack_name = "myproject"
s3_bucket = "..........."
s3_prefix = "myproject"
region = "us-east-2"
confirm_changeset = true
capabilities = ["CAPABILITY_IAM", "CAPABILITY_NAMED_IAM"]
parameter_overrides = "EnvironmentName=\"TEST\""
image_repositories = []
template_file = "template_test.yaml"

[TEST.validate.parameters]
template_file = "template_test.yaml"
region = "us-east-2"
lint = true

[TEST.build.parameters]
use_container = true
template_file = "template_test.yaml"
```

Also, very frustratingly, I am having a really hard time trying to install the older versions on homebrew. Running `brew install aws-sam-cli@1.76.0` doesn't work; it can't find it at all.

### Steps to reproduce:
Run `sam deploy --config-env TEST` (test is the name of my given deployment stack version)

Building and validating works fine though.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)
macOS Ventura 13.3

**Comment by mikegrima** [comment]
I was able to downgrade by uninstalling it from homebrew, and installing it in a Python virtual environment. I needed to install `1.76.0`. `1.77.0` is broken as well.

**Comment by mikegrima** [comment]
This may be a duplicate of #4934

**Comment by FireAndIceFrog** [comment]
I am having this issue too, we use the latest sam cli.

**Comment by moelasmar** [comment]
thanks @mikegrima for raising this issue. I can reproduce this issue.
Could you please try to change the `capabilities = ["CAPABILITY_IAM", "CAPABILITY_NAMED_IAM"]` line to be `capabilities = "CAPABILITY_IAM CAPABILITY_NAMED_IAM"`, and let us know if it solves your issue. 

I just need to double check if it is related to capabilities attribute only, or it is related to any other attribute

**Comment by moelasmar** [comment]
mainly tomlkit version `0.11.6` cause this issue, I tried version `0.11.7` does not have this issue.

**Comment by mikegrima** [comment]
@moelasmar : I have confirmed that making the change to `capabilities = "CAPABILITY_IAM CAPABILITY_NAMED_IAM"` works.

**Comment by qingchm** [comment]
Hey @mikegrima  this has been addressed in our most recent release 1.79.0 (https://github.com/aws/aws-sam-cli/releases/tag/v1.79.0). Closing the issue here. Thanks for the engagement!
